vehicle-python/pyproject.toml

```diff
@@ -101,6 +108,31 @@ testpaths = ["tests"]
 [tool.setuptools.dynamic]
 version = { attr = "vehicle_lang._version.VERSION" }
 
+[tool.uv]
+default-groups = []
+
+[dependency-groups]
+test = [
+  "pytest>=7.1,<9",
+  "packaging>=23",
+  "pygments>=2.14,<3",
+  "numpy>=1.21,<3",
+]
+mypy = [
+  "types-setuptools>=45",
+  "types-pygments>=2.14,<3",
+]
+pygments = ["pygments>=2.14,<3"]
+numpy = ["numpy>=1.21,<3"]
+jax = ["jax>=0.4.26"]
+dev = ["pre-commit>=4.3.0"]
+tensorflow = [
+  "tensorflow>=2,<3; (python_version >= '3.10' and python_version < '3.13') and ((sys_platform == 'linux' and (platform_machine == 'x86_64' or platform_machine == 'aarch64')) or (sys_platform == 'darwin' and platform_machine == 'x86_64') or (sys_platform == 'win32' and platform_machine == 'AMD64'))",
```

Comment:
Do we really need to duplicate all this stuff here? If so, can you add a comment explaining why?